### PR TITLE
Modified how interval bounds work

### PR DESCRIPTION
**Pull request by ClaireDons**
Things added/changed:

- Option for "greedy" mode which creates an interval is any data is available within the time bound
- Create an interval while inferring data frequency in "safe" mode
- Tests for these changes and modified some existing tests

### CODE REVIEW

**Comment by BSchilperoort** [comment]
Can you update the changelog? I think this PR is basically ready now!

**Review by BSchilperoort — APPROVED** [review]
Wrote this PR with some pair-programming already, so I won't go over everything in detail.

I think only a changelog entry is remaining, and then this can be merged!

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,15 +6,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
...
+ - Create an interval while inferring data frequency in "safe" mode ([#75](https://github.com/AI4S2S/lilio/pull/75))
+ - Tests for greedy and safe mode, including inferring data frequency ([#75](https://github.com/AI4S2S/lilio/pull/75))
+
 
 ### Fixed
  - Fixed issue with calendar generation when the (rightmost) target period crossed into the new year ([#70](https://github.com/AI4S2S/lilio/pull/70)).
  - Fixed issue with Pandas 2.2 where 'M' changed to 'ME' ([#72](https://github.com/AI4S2S/lilio/pull/72))
  - Fixed issue where pd.Interval now shows time where it is not wanted ([#72](https://github.com/AI4S2S/lilio/pull/72)). The issue was raised with Pandas ([#57748] https://github.com/pandas-dev/pandas/issues/57748).
  - Added tests for latest version of macos (with M1 chips) ([#73](https://github.com/AI4S2S/lilio/pull/73))
+ - Fixed dummy timeseries, incorrect dates and test data, which should not have been working before ([#75](https://github.com/AI4S2S/lilio/pull/75))
```
Comment (BSchilperoort) [inline CHANGELOG.md:23]:
```suggestion
 - Fixed a bug in `map_to_data` that lead to anchor years being included when they should not have ([#75](https://github.com/AI4S2S/lilio/pull/75))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,15 +6,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
 
 ## [Unreleased]
 
+## 0.5.0 (2024-06-11)
 ### Changed
  - Moved to ruff formatter instead of black ([#70](https://github.com/AI4S2S/lilio/pull/70))
  - Do import sorting with ruff instead of isort ([#70](https://github.com/AI4S2S/lilio/pull/70))
+ - Option for "greedy" mode which creates an interval is any data is available within the time bound ([#75](https://github.com/AI4S2S/lilio/pull/75))
```
Comment (BSchilperoort) [inline CHANGELOG.md:13]:
```suggestion
 - Option for "greedy" mode which creates an interval if any data is available within the time bound ([#75](https://github.com/AI4S2S/lilio/pull/75))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,15 +6,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
...
+ - Create an interval while inferring data frequency in "safe" mode ([#75](https://github.com/AI4S2S/lilio/pull/75))
```
Comment (BSchilperoort) [inline CHANGELOG.md:14]:
```suggestion
 - `map_to_data` now tries to infer the time bounds for the data, and uses those to determine which anchor years are valid ([#75](https://github.com/AI4S2S/lilio/pull/75))
```

**Review by BSchilperoort — APPROVED** [review]
Just some comments on the changelog. Looks good otherwise

File: CHANGELOG.md (hunk 3 above)
Comment (ClaireDons) [reply CHANGELOG.md]: Thanks, those are comments are clearer!

**Pull request merged**